File: snippets/snippets/node/embed-post/embed-image-post.ts
const { CohereClient } = require('cohere-ai');

const cohere = new CohereClient({});

(async () => {
  const image = await fetch('https://cohere.com/favicon-32x32.png');
  const buffer = await image.arrayBuffer();
  const stringifiedBuffer = Buffer.from(buffer).toString('base64');
  const contentType = image.headers.get('content-type');
  const imageBase64 = `data:${contentType};base64,${stringifiedBuffer}`;

  const embed = await cohere.embed({
    model: 'embed-v4.0',
    inputType: 'image',
    embeddingTypes: ['float'],
    images: [imageBase64],
  });
  console.log(embed);
})();

File: snippets/snippets/python/embed-post/embed-image-post.py
import cohere
import requests
import base64

co = cohere.Client()

image = requests.get("https://cohere.com/favicon-32x32.png")
stringified_buffer = base64.b64encode(image.content).decode("utf-8")
content_type = image.headers["Content-Type"]
image_base64 = f"data:{content_type};base64,{stringified_buffer}"

response = co.embed(
    model="embed-english-v3.0",
    input_type="image",
    embedding_types=["float"],
    images=[image_base64],
)

print(response)
